src/deriver/fact_saver.py

```
@@ -0,0 +1,281 @@
+        # Start worker tasks
+        for i in range(self.max_workers):
+            worker = asyncio.create_task(self._worker(f"worker-{i}"))
+            self.workers.append(worker)
+            
+        logger.info("Observation saver started successfully")
+
+    async def stop(self):
+        """Stop all workers and wait for completion."""
+        if not self.is_running:
+            return
+            
```

Pylint (3.3.7)

[convention] Trailing whitespace

(C0303)